### ISSUE TITLE
[wish list] publish package on pypi

### ISSUE THREAD

**Issue by alicia-ziying-yang** [alicia-ziying-yang/conTEXT-explorer#16]
As suggested in #15 , should we publish the current version to pypi?

**Comment by faroit** [comment]
I think pypi deployment is generally a good idea as this would simply installation for users.

publishing to pypi is not part of the JOSS review guidelines right now, but it will certainly be added soon: https://github.com/openjournals/joss/pull/976

https://github.com/openjournals/joss-reviews/issues/3347

**Comment by alicia-ziying-yang** [comment]
built and uploaded for test here:
https://test.pypi.org/project/ConTEXT-Explorer/
But, the user need to install all required packages by themselves (otherwise cannot install), plus run 
`python -m spacy download en` before starting the app.

For me, git clone this repo, and install via setup.py (`pip install . `) is much better though.
@turpinandrew what do you think?

**Comment by faroit** [comment]
@alicia-ziying-yang

> But, the user need to install all required packages by themselves (otherwise cannot install)

the dependencies should be added to the `install_requires` section of setup.py to make this work. I know this is redundant, but it would really make the installation a one-liner then... (assuming the right python version ;-) )

> plus run python -m spacy download en before starting the app. 

wouldn't [this](https://stackoverflow.com/a/55704556/479153) workaround work?

**Comment by alicia-ziying-yang** [comment]
> the dependencies should be added to the install_requires section of setup.py to make this work. I know this is redundant, but it would really make the installation a one-liner then... (assuming the right python version ;-) )

They had been there (in the setup.py) since the first time. And if run `pip install .` which runs the setup.py, it installs 'python -m spacy download en'. That are the steps in my README.md. It works well.

But if install by pypi, I will get these errors if I create a clean venv for testing:
`% pip install -i https://test.pypi.org/simple/ ConTEXT-Explorer==1.0.1 
Looking in indexes: https://test.pypi.org/simple/
Collecting ConTEXT-Explorer==1.0.1
 Downloading https://test-files.pythonhosted.org/packages/75/e6/09322a3c6c16f41c56597510cbcbb5c4a9afb96d73f0073e82bc3d8dabd4/ConTEXT_Explorer-1.0.1-py3-none-any.whl (35 kB)
ERROR: Could not find a version that satisfies the requirement dash-core-components==1.10.2 (from context-explorer) (from versions: none)
ERROR: No matching distribution found for dash-core-components==1.10.2`

until, I install everything in the requirements by hand

**Comment by alicia-ziying-yang** [comment]
I can attach how I build the package by following the official guide (https://packaging.python.org/tutorials/packaging-projects/):

% python3 -m build
* Creating venv isolated environment...
* Installing packages in isolated environment... (pip, setuptools)
* Getting dependencies for sdist...
WARNING: '.' not a valid package name; please use only .-separated package names in setup.py
running egg_info
creating ConTEXT_Explorer.egg-info
writing ConTEXT_Explorer.egg-info/PKG-INFO
writing dependency_links to ConTEXT_Explorer.egg-info/dependency_links.txt
writing entry points to ConTEXT_Explorer.egg-info/entry_points.txt
writing requirements to ConTEXT_Explorer.egg-info/requires.txt
writing top-level names to ConTEXT_Explorer.egg-info/top_level.txt
writing manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
package init file './__init__.py' not found (or not a regular file)
package init file 'whoosh_search/__init__.py' not found (or not a regular file)
package init file 'topic_model/__init__.py' not found (or not a regular file)
reading manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
* Building sdist...
WARNING: '.' not a valid package name; please use only .-separated package names in setup.py
running sdist
running egg_info
writing ConTEXT_Explorer.egg-info/PKG-INFO
writing dependency_links to ConTEXT_Explorer.egg-info/dependency_links.txt
writing entry points to ConTEXT_Explorer.egg-info/entry_points.txt
writing requirements to ConTEXT_Explorer.egg-info/requires.txt
writing top-level names to ConTEXT_Explorer.egg-info/top_level.txt
package init file './__init__.py' not found (or not a regular file)
package init file 'whoosh_search/__init__.py' not found (or not a regular file)
package init file 'topic_model/__init__.py' not found (or not a regular file)
reading manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
running check
creating ConTEXT-Explorer-1.0.2
creating ConTEXT-Explorer-1.0.2/ConTEXT_Explorer.egg-info
creating ConTEXT-Explorer-1.0.2/topic_model
creating ConTEXT-Explorer-1.0.2/whoosh_search
copying files to ConTEXT-Explorer-1.0.2...
copying LICENSE -> ConTEXT-Explorer-1.0.2
copying README.md -> ConTEXT-Explorer-1.0.2
copying pyproject.toml -> ConTEXT-Explorer-1.0.2
copying setup.py -> ConTEXT-Explorer-1.0.2
copying ./app.py -> ConTEXT-Explorer-1.0.2/.
copying ./test_cases.py -> ConTEXT-Explorer-1.0.2/.
copying ConTEXT_Explorer.egg-info/PKG-INFO -> ConTEXT-Explorer-1.0.2/ConTEXT_Explorer.egg-info
copying ConTEXT_Explorer.egg-info/SOURCES.txt -> ConTEXT-Explorer-1.0.2/ConTEXT_Explorer.egg-info
copying ConTEXT_Explorer.egg-info/dependency_links.txt -> ConTEXT-Explorer-1.0.2/ConTEXT_Explorer.egg-info
copying ConTEXT_Explorer.egg-info/entry_points.txt -> ConTEXT-Explorer-1.0.2/ConTEXT_Explorer.egg-info
copying ConTEXT_Explorer.egg-info/requires.txt -> ConTEXT-Explorer-1.0.2/ConTEXT_Explorer.egg-info
copying ConTEXT_Explorer.egg-info/top_level.txt -> ConTEXT-Explorer-1.0.2/ConTEXT_Explorer.egg-info
copying topic_model/generate_models_fromapp.py -> ConTEXT-Explorer-1.0.2/topic_model
copying topic_model/word2vec.py -> ConTEXT-Explorer-1.0.2/topic_model
copying whoosh_search/preprocess_corpus.py -> ConTEXT-Explorer-1.0.2/whoosh_search
Writing ConTEXT-Explorer-1.0.2/setup.cfg
Creating tar archive
removing 'ConTEXT-Explorer-1.0.2' (and everything under it)
* Building wheel from sdist
* Creating venv isolated environment...
* Installing packages in isolated environment... (pip, setuptools)
* Getting dependencies for wheel...
WARNING: '.' not a valid package name; please use only .-separated package names in setup.py
running egg_info
writing ConTEXT_Explorer.egg-info/PKG-INFO
writing dependency_links to ConTEXT_Explorer.egg-info/dependency_links.txt
writing entry points to ConTEXT_Explorer.egg-info/entry_points.txt
writing requirements to ConTEXT_Explorer.egg-info/requires.txt
writing top-level names to ConTEXT_Explorer.egg-info/top_level.txt
package init file './__init__.py' not found (or not a regular file)
package init file 'whoosh_search/__init__.py' not found (or not a regular file)
package init file 'topic_model/__init__.py' not found (or not a regular file)
reading manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
* Installing packages in isolated environment... (wheel)
* Building wheel...
WARNING: '.' not a valid package name; please use only .-separated package names in setup.py
running bdist_wheel
running build
running build_py
package init file './__init__.py' not found (or not a regular file)
creating build
creating build/lib
copying ./setup.py -> build/lib
copying ./app.py -> build/lib
copying ./test_cases.py -> build/lib
package init file 'whoosh_search/__init__.py' not found (or not a regular file)
creating build/lib/whoosh_search
copying whoosh_search/preprocess_corpus.py -> build/lib/whoosh_search
package init file 'topic_model/__init__.py' not found (or not a regular file)
creating build/lib/topic_model
copying topic_model/word2vec.py -> build/lib/topic_model
copying topic_model/generate_models_fromapp.py -> build/lib/topic_model
running egg_info
writing ConTEXT_Explorer.egg-info/PKG-INFO
writing dependency_links to ConTEXT_Explorer.egg-info/dependency_links.txt
writing entry points to ConTEXT_Explorer.egg-info/entry_points.txt
writing requirements to ConTEXT_Explorer.egg-info/requires.txt
writing top-level names to ConTEXT_Explorer.egg-info/top_level.txt
reading manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'ConTEXT_Explorer.egg-info/SOURCES.txt'
running CustomCommands
Running: ['pip', 'install', 'spacy==2.2.4']
Output: b'WARNING: The directory \'/Users/ziyingy3/Library/Caches/pip/http\' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo\'s -H flag.\nWARNING: The directory \'/Users/ziyingy3/Library/Caches/pip\' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo\'s -H flag.\nCollecting spacy==2.2.4\n  Downloading https://files.pythonhosted.org/packages/a7/90/785efc7bd26ff8e399f03d02b259216cf29b389a8f3c2412624f0ac32b00/spacy-2.2.4-cp37-cp37m-macosx_10_9_x86_64.whl (10.5MB)\nCollecting plac<1.2.0,>=0.9.6 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/86/85/40b8f66c2dd8f4fd9f09d59b22720cffecf1331e788b8a0cab5bafb353d1/plac-1.1.3-py2.py3-none-any.whl\nCollecting thinc==7.4.0 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/1c/4b/6b49c6e576890b806fab44c2983f3dc1723d38accb2c2193a8da15e88ffb/thinc-7.4.0-cp37-cp37m-macosx_10_9_x86_64.whl (2.2MB)\nCollecting murmurhash<1.1.0,>=0.28.0 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/17/34/5abf7fcb3085dc7ad752af9a9b517d739ce62d16061f97cdda7bce0b53fc/murmurhash-1.0.5-cp37-cp37m-macosx_10_9_x86_64.whl\nCollecting wasabi<1.1.0,>=0.4.0 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/a6/1d/d281571b4c3b20fff183b485c6673c62878727119a849c7932651a8b5060/wasabi-0.8.2-py3-none-any.whl\nCollecting numpy>=1.15.0 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/e0/54/49206c6aba7d0e7d5a6e2f5b6760a00e35bb705e9d2be007be0a16e4ba40/numpy-1.21.2-cp37-cp37m-macosx_10_9_x86_64.whl (16.9MB)\nCollecting tqdm<5.0.0,>=4.38.0 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/63/f3/b7a1b8e40fd1bd049a34566eb353527bb9b8e9b98f8b6cf803bb64d8ce95/tqdm-4.62.3-py2.py3-none-any.whl (76kB)\nCollecting blis<0.5.0,>=0.4.0 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/85/d8/f0be9d8ebec9cbeea1427de6ac0ecc919c0bfe881eff2d2965dbc310ca8b/blis-0.4.1-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (4.0MB)\nRequirement already satisfied: setuptools in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy==2.2.4) (58.1.0)\nCollecting cymem<2.1.0,>=2.0.2 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/f5/e4/270d7d5b64528480f6f9923325266486ab286c014ce83610d7bc43368218/cymem-2.0.5-cp37-cp37m-macosx_10_9_x86_64.whl\nCollecting requests<3.0.0,>=2.13.0 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/92/96/144f70b972a9c0eabbd4391ef93ccd49d0f2747f4f6a2a2738e99e5adc65/requests-2.26.0-py2.py3-none-any.whl (62kB)\nCollecting srsly<1.1.0,>=1.0.2 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/85/be/1b94141d3d65ddf86a7e63daeba72fbbf054ad6ad3a5d42b296f68c70030/srsly-1.0.5-cp37-cp37m-macosx_10_9_x86_64.whl (177kB)\nCollecting catalogue<1.1.0,>=0.0.7 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/6c/f9/9a5658e2f56932e41eb264941f9a2cb7f3ce41a80cb36b2af6ab78e2f8af/catalogue-1.0.0-py2.py3-none-any.whl\nCollecting preshed<3.1.0,>=3.0.2 (from spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/f2/10/f18af885d80d458eaa25732348f70dbb1b83d24fa49a9fb5fee963f24def/preshed-3.0.5-cp37-cp37m-macosx_10_9_x86_64.whl (104kB)\nCollecting certifi>=2017.4.17 (from requests<3.0.0,>=2.13.0->spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/05/1b/0a0dece0e8aa492a6ec9e4ad2fe366b511558cdc73fd3abc82ba7348e875/certifi-2021.5.30-py2.py3-none-any.whl (145kB)\nCollecting urllib3<1.27,>=1.21.1 (from requests<3.0.0,>=2.13.0->spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/af/f4/524415c0744552cce7d8bf3669af78e8a069514405ea4fcbd0cc44733744/urllib3-1.26.7-py2.py3-none-any.whl (138kB)\nCollecting idna<4,>=2.5; python_version >= "3" (from requests<3.0.0,>=2.13.0->spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/d7/77/ff688d1504cdc4db2a938e2b7b9adee5dd52e34efbd2431051efc9984de9/idna-3.2-py3-none-any.whl (59kB)\nCollecting charset-normalizer~=2.0.0; python_version >= "3" (from requests<3.0.0,>=2.13.0->spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/3f/65/69e6754102dcd018a0f29e4db673372eb323ee504431125ab6c9109cb21c/charset_normalizer-2.0.6-py3-none-any.whl\nCollecting importlib-metadata>=0.20; python_version < "3.8" (from catalogue<1.1.0,>=0.0.7->spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/71/c2/cb1855f0b2a0ae9ccc9b69f150a7aebd4a8d815bd951e74621c4154c52a8/importlib_metadata-4.8.1-py3-none-any.whl\nCollecting zipp>=0.5 (from importlib-metadata>=0.20; python_version < "3.8"->catalogue<1.1.0,>=0.0.7->spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/bd/df/d4a4974a3e3957fd1c1fa3082366d7fff6e428ddb55f074bf64876f8e8ad/zipp-3.6.0-py3-none-any.whl\nCollecting typing-extensions>=3.6.4; python_version < "3.8" (from importlib-metadata>=0.20; python_version < "3.8"->catalogue<1.1.0,>=0.0.7->spacy==2.2.4)\n  Downloading https://files.pythonhosted.org/packages/74/60/18783336cc7fcdd95dae91d73477830aa53f5d3181ae4fe20491d7fc3199/typing_extensions-3.10.0.2-py3-none-any.whl\nInstalling collected packages: plac, murmurhash, numpy, blis, cymem, tqdm, zipp, typing-extensions, importlib-metadata, catalogue, preshed, srsly, wasabi, thinc, certifi, urllib3, idna, charset-normalizer, requests, spacy\nSuccessfully installed blis-0.4.1 catalogue-1.0.0 certifi-2021.5.30 charset-normalizer-2.0.6 cymem-2.0.5 idna-3.2 importlib-metadata-4.8.1 murmurhash-1.0.5 numpy-1.21.2 plac-1.1.3 preshed-3.0.5 requests-2.26.0 spacy-2.2.4 srsly-1.0.5 thinc-7.4.0 tqdm-4.62.3 typing-extensions-3.10.0.2 urllib3-1.26.7 wasabi-0.8.2 zipp-3.6.0\nWARNING: You are using pip version 19.2.3, however version 21.2.4 is available.\nYou should consider upgrading via the \'pip install --upgrade pip\' command.\n'
Running: ['python', '-m', 'spacy', 'download', 'en']
Output: b'Collecting en_core_web_sm==2.2.5 from https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.2.5/en_core_web_sm-2.2.5.tar.gz#egg=en_core_web_sm==2.2.5\n  Downloading https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.2.5/en_core_web_sm-2.2.5.tar.gz (12.0MB)\nRequirement already satisfied: spacy>=2.2.2 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from en_core_web_sm==2.2.5) (2.2.4)\nRequirement already satisfied: numpy>=1.15.0 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (1.21.2)\nRequirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (1.0.5)\nRequirement already satisfied: requests<3.0.0,>=2.13.0 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (2.26.0)\nRequirement already satisfied: tqdm<5.0.0,>=4.38.0 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (4.62.3)\nRequirement already satisfied: preshed<3.1.0,>=3.0.2 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (3.0.5)\nRequirement already satisfied: plac<1.2.0,>=0.9.6 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (1.1.3)\nRequirement already satisfied: wasabi<1.1.0,>=0.4.0 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (0.8.2)\nRequirement already satisfied: catalogue<1.1.0,>=0.0.7 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (1.0.0)\nRequirement already satisfied: thinc==7.4.0 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (7.4.0)\nRequirement already satisfied: srsly<1.1.0,>=1.0.2 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (1.0.5)\nRequirement already satisfied: blis<0.5.0,>=0.4.0 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (0.4.1)\nRequirement already satisfied: setuptools in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (58.1.0)\nRequirement already satisfied: cymem<2.1.0,>=2.0.2 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from spacy>=2.2.2->en_core_web_sm==2.2.5) (2.0.5)\nRequirement already satisfied: idna<4,>=2.5; python_version >= "3" in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.2->en_core_web_sm==2.2.5) (3.2)\nRequirement already satisfied: certifi>=2017.4.17 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.2->en_core_web_sm==2.2.5) (2021.5.30)\nRequirement already satisfied: charset-normalizer~=2.0.0; python_version >= "3" in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.2->en_core_web_sm==2.2.5) (2.0.6)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.2->en_core_web_sm==2.2.5) (1.26.7)\nRequirement already satisfied: importlib-metadata>=0.20; python_version < "3.8" in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from catalogue<1.1.0,>=0.0.7->spacy>=2.2.2->en_core_web_sm==2.2.5) (4.8.1)\nRequirement already satisfied: zipp>=0.5 in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from importlib-metadata>=0.20; python_version < "3.8"->catalogue<1.1.0,>=0.0.7->spacy>=2.2.2->en_core_web_sm==2.2.5) (3.6.0)\nRequirement already satisfied: typing-extensions>=3.6.4; python_version < "3.8" in /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages (from importlib-metadata>=0.20; python_version < "3.8"->catalogue<1.1.0,>=0.0.7->spacy>=2.2.2->en_core_web_sm==2.2.5) (3.10.0.2)\nBuilding wheels for collected packages: en-core-web-sm\n  Building wheel for en-core-web-sm (setup.py): started\n  Building wheel for en-core-web-sm (setup.py): finished with status \'done\'\n  Created wheel for en-core-web-sm: filename=en_core_web_sm-2.2.5-cp37-none-any.whl size=12011742 sha256=46df924fa91b679d611136e4b96a98205b17f56386607701ffd4ea00229eca88\n  Stored in directory: /private/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/pip-ephem-wheel-cache-3xsodgnw/wheels/6a/47/fb/6b5a0b8906d8e8779246c67d4658fd8a544d4a03a75520197a\nSuccessfully built en-core-web-sm\nInstalling collected packages: en-core-web-sm\nSuccessfully installed en-core-web-sm-2.2.5\nWARNING: You are using pip version 19.2.3, however version 21.2.4 is available.\nYou should consider upgrading via the \'pip install --upgrade pip\' command.\n\x1b[38;5;2m\xe2\x9c\x94 Download and installation successful\x1b[0m\nYou can now load the model via spacy.load(\'en_core_web_sm\')\n\x1b[38;5;2m\xe2\x9c\x94 Linking successful\x1b[0m\n/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages/en_core_web_sm\n-->\n/var/folders/kj/ttgtj5fx2bv3xxhbc9vk37km0000gp/T/build-env-fszaaewx/lib/python3.7/site-packages/spacy/data/en\nYou can now load the model via spacy.load(\'en\')\n'
installing to build/bdist.macosx-10.7-x86_64/wheel
running install
running install_lib
creating build/bdist.macosx-10.7-x86_64
creating build/bdist.macosx-10.7-x86_64/wheel
creating build/bdist.macosx-10.7-x86_64/wheel/topic_model
copying build/lib/topic_model/word2vec.py -> build/bdist.macosx-10.7-x86_64/wheel/topic_model
copying build/lib/topic_model/generate_models_fromapp.py -> build/bdist.macosx-10.7-x86_64/wheel/topic_model
copying build/lib/setup.py -> build/bdist.macosx-10.7-x86_64/wheel
copying build/lib/app.py -> build/bdist.macosx-10.7-x86_64/wheel
creating build/bdist.macosx-10.7-x86_64/wheel/whoosh_search
copying build/lib/whoosh_search/preprocess_corpus.py -> build/bdist.macosx-10.7-x86_64/wheel/whoosh_search
copying build/lib/test_cases.py -> build/bdist.macosx-10.7-x86_64/wheel
running install_egg_info
Copying ConTEXT_Explorer.egg-info to build/bdist.macosx-10.7-x86_64/wheel/ConTEXT_Explorer-1.0.2-py3.7.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
creating build/bdist.macosx-10.7-x86_64/wheel/ConTEXT_Explorer-1.0.2.dist-info/WHEEL
creating '/Users/ziyingy3/Desktop/ce/org/conTEXT-explorer/dist/tmps4b637in/ConTEXT_Explorer-1.0.2-py3-none-any.whl' and adding 'build/bdist.macosx-10.7-x86_64/wheel' to it
adding 'app.py'
adding 'setup.py'
adding 'test_cases.py'
adding 'topic_model/generate_models_fromapp.py'
adding 'topic_model/word2vec.py'
adding 'whoosh_search/preprocess_corpus.py'
adding 'ConTEXT_Explorer-1.0.2.dist-info/LICENSE'
adding 'ConTEXT_Explorer-1.0.2.dist-info/METADATA'
adding 'ConTEXT_Explorer-1.0.2.dist-info/WHEEL'
adding 'ConTEXT_Explorer-1.0.2.dist-info/entry_points.txt'
adding 'ConTEXT_Explorer-1.0.2.dist-info/top_level.txt'
adding 'ConTEXT_Explorer-1.0.2.dist-info/RECORD'
removing build/bdist.macosx-10.7-x86_64/wheel
Successfully built ConTEXT-Explorer-1.0.2.tar.gz and ConTEXT_Explorer-1.0.2-py3-none-any.whl

% python3 -m pip install --upgrade twine
  Downloading twine-3.4.2-py3-none-any.whl (34 kB)
Collecting colorama>=0.4.3
  Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting tqdm>=4.14
  Downloading tqdm-4.62.3-py2.py3-none-any.whl (76 kB)
     |████████████████████████████████| 76 kB 6.5 MB/s 
Collecting readme-renderer>=21.0
  Downloading readme_renderer-29.0-py2.py3-none-any.whl (15 kB)
Requirement already satisfied: importlib-metadata>=3.6 in /Users/ziyingy3/opt/anaconda3/envs/ce-test2/lib/python3.7/site-packages (from twine) (4.8.1)
Collecting requests>=2.20
  Downloading requests-2.26.0-py2.py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 12.9 MB/s 
Collecting rfc3986>=1.4.0
  Downloading rfc3986-1.5.0-py2.py3-none-any.whl (31 kB)
Collecting requests-toolbelt!=0.9.0,>=0.8.0
  Downloading requests_toolbelt-0.9.1-py2.py3-none-any.whl (54 kB)
     |████████████████████████████████| 54 kB 10.7 MB/s 
Collecting keyring>=15.1
  Downloading keyring-23.2.1-py3-none-any.whl (33 kB)
Collecting pkginfo>=1.4.2
  Downloading pkginfo-1.7.1-py2.py3-none-any.whl (25 kB)
Requirement already satisfied: typing-extensions>=3.6.4 in /Users/ziyingy3/opt/anaconda3/envs/ce-test2/lib/python3.7/site-packages (from importlib-metadata>=3.6->twine) (3.10.0.2)
Requirement already satisfied: zipp>=0.5 in /Users/ziyingy3/opt/anaconda3/envs/ce-test2/lib/python3.7/site-packages (from importlib-metadata>=3.6->twine) (3.6.0)
Collecting six
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting docutils>=0.13.1
  Downloading docutils-0.17.1-py2.py3-none-any.whl (575 kB)
     |████████████████████████████████| 575 kB 6.4 MB/s 
Collecting bleach>=2.1.0
  Downloading bleach-4.1.0-py2.py3-none-any.whl (157 kB)
     |████████████████████████████████| 157 kB 7.4 MB/s 
Collecting Pygments>=2.5.1
  Downloading Pygments-2.10.0-py3-none-any.whl (1.0 MB)
     |████████████████████████████████| 1.0 MB 8.6 MB/s 
Requirement already satisfied: packaging in /Users/ziyingy3/opt/anaconda3/envs/ce-test2/lib/python3.7/site-packages (from bleach>=2.1.0->readme-renderer>=21.0->twine) (21.0)
Collecting webencodings
  Downloading webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting charset-normalizer~=2.0.0
  Downloading charset_normalizer-2.0.6-py3-none-any.whl (37 kB)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.7-py2.py3-none-any.whl (138 kB)
     |████████████████████████████████| 138 kB 6.4 MB/s 
Requirement already satisfied: certifi>=2017.4.17 in /Users/ziyingy3/opt/anaconda3/envs/ce-test2/lib/python3.7/site-packages (from requests>=2.20->twine) (2021.5.30)
Collecting idna<4,>=2.5
  Downloading idna-3.2-py3-none-any.whl (59 kB)
     |████████████████████████████████| 59 kB 8.2 MB/s 
Requirement already satisfied: pyparsing>=2.0.2 in /Users/ziyingy3/opt/anaconda3/envs/ce-test2/lib/python3.7/site-packages (from packaging->bleach>=2.1.0->readme-renderer>=21.0->twine) (2.4.7)
Installing collected packages: webencodings, urllib3, six, idna, charset-normalizer, requests, Pygments, docutils, bleach, tqdm, rfc3986, requests-toolbelt, readme-renderer, pkginfo, keyring, colorama, twine
Successfully installed Pygments-2.10.0 bleach-4.1.0 charset-normalizer-2.0.6 colorama-0.4.4 docutils-0.17.1 idna-3.2 keyring-23.2.1 pkginfo-1.7.1 readme-renderer-29.0 requests-2.26.0 requests-toolbelt-0.9.1 rfc3986-1.5.0 six-1.16.0 tqdm-4.62.3 twine-3.4.2 urllib3-1.26.7 webencodings-0.5.1
(ce-test2) ziyingy3@9999L-206579-M conTEXT-explorer % python3 -m twine upload --repository testpypi dist/*
Uploading distributions to https://test.pypi.org/legacy/
Uploading ConTEXT_Explorer-1.0.2-py3-none-any.whl
100%|███████████████████████████████████| 40.0k/40.0k [00:02<00:00, 17.1kB/s]
Uploading ConTEXT-Explorer-1.0.2.tar.gz
100%|███████████████████████████████████| 36.8k/36.8k [00:02<00:00, 16.6kB/s]

View at:
https://test.pypi.org/project/ConTEXT-Explorer/1.0.2/


So if I try  `% pip install -i https://test.pypi.org/simple/ ConTEXT-Explorer==1.0.2`
It gives me ERROR like:
```
ERROR: Could not find a version that satisfies the requirement dash-uploader==0.4.1 (from context-explorer) (from versions: none)
ERROR: No matching distribution found for dash-uploader==0.4.1
```
for random package in the requirement.

I do not understand what's wrong. Have googled for a whole day, many people have this issue but for now I have not found any solution.

**Comment by alicia-ziying-yang** [comment]
The latest upload: https://test.pypi.org/project/ConTEXT-Explorer/1.0.2/
But I cannot install it via `% pip install -i https://test.pypi.org/simple/ ConTEXT-Explorer==1.0.2` though.

**Comment by alicia-ziying-yang** [comment]
Clearly, pypi does not run "setup.py" after downloading.

Also a new problem: even successfully install it into python site-package via pypi, user cannot upload new corpus because of path problem:
`FileNotFoundError: [Errno 2] No such file or directory: './topic_model/sample_data_2/processed_content_sample_data_2.pkl'`
it cannot write file into the sys path, even I already use "sudo".

This is because this application was not designed for system package use, but a stand-alone Application. Of course we can further improve it to archive that, but we now can only put it into [wish list] because of time limitation. @turpinandrew ?

@faroit Thanks for suggesting to make it a pypi package. But for now I still recommend people to install it via the method via git clone, and `pip install . ` (shown in README.md) which handles everything.

**Comment by turpinandrew** [comment]
> Clearly, pypi does not run "setup.py" after downloading.
> 
> Also a new problem: even successfully install it into python site-package via pypi, user cannot upload new corpus because of path problem: `FileNotFoundError: [Errno 2] No such file or directory: './topic_model/sample_data_2/processed_content_sample_data_2.pkl'` it cannot write file into the sys path, even I already use "sudo".
> 
> This is because this application was not designed for system package use, but a stand-alone Application. Of course we can further improve it to archive that, but we now can only put it into [wish list] because of time limitation. @turpinandrew ?
> 
> @faroit Thanks for suggesting to make it a pypi package. But for now I still recommend people to install it via the method via git clone, and `pip install . ` (shown in README.md) which handles everything.

**Comment by turpinandrew** [comment]
Doh - think I closed it, which is probably rude! 
I agree, Alicia, this was never intended to be a 'package', but a standalone application. It seems a bit odd to jump through hoops to have it on pypi.
If there is a two step process (download, run setup.py), that would seem appropriate for an application?

**Comment by faroit** [comment]
@turpinandrew @alicia-ziying-yang thanks for trying. It looks indeed like to be a lot of a work and I value that you tried this. Maybe as a compromise you could add a conda environment.yaml (+documentation in readme) file that includes the python version thus making it simpler to recreate the environment from scratch.  

__Update__ I saw you already did this

**Comment by faroit** [comment]
@turpinandrew can be closed then

**Comment by alicia-ziying-yang** [comment]
> @turpinandrew @alicia-ziying-yang thanks for trying. It looks indeed like to be a lot of a work and I value that you tried this. Maybe as a compromise you could add a conda environment.yaml (+documentation in readme) file that includes the python version thus making it simpler to recreate the environment from scratch.
> 
> **Update** I saw you already did this

Hi @faroit @turpinandrew , thank you very much.
I have updated the conda env file, and added instructions in README.md. Please check, and @turpinandrew can close this issue.

**Comment by turpinandrew** [comment]
Thanks for all your efforts @alicia-ziying-yang.
